src/13_framebuffers.rs

```diff
@@ -0,0 +1,518 @@
+extern crate env_logger;
+#[cfg(feature = "dx12")]
+extern crate gfx_backend_dx12 as back;
+#[cfg(feature = "metal")]
+extern crate gfx_backend_metal as back;
+#[cfg(feature = "vulkan")]
+extern crate gfx_backend_vulkan as back;
+extern crate gfx_hal as hal;
+extern crate glsl_to_spirv;
+extern crate winit;
+
+use hal::{Capability, Device, Instance, PhysicalDevice, QueueFamily, Surface, SwapchainConfig};
+use std::io::Read;
+
+static WINDOW_NAME: &str = "13_framebuffers";
+
+fn main() {
+    env_logger::init();
+    let (window, events_loop) = init_window();
+
+    let (
+        _instance,
+        device,
+        swapchain,
+        _surface,
+        frame_images,
+        render_pass,
+        descriptor_set_layouts,
+        pipeline_layout,
+        gfx_pipeline,
+        swapchain_framebuffers,
+    ) = init_hal(&window);
+    main_loop(events_loop);
+
+    clean_up(
+        device,
+        frame_images,
+        swapchain,
+        render_pass,
+        descriptor_set_layouts,
+        pipeline_layout,
+        gfx_pipeline,
+        swapchain_framebuffers,
+    );
+}
+
+fn create_framebuffers(
+    device: &<back::Backend as hal::Backend>::Device,
+    render_pass: &<back::Backend as hal::Backend>::RenderPass,
+    frame_images: &Vec<(
+        <back::Backend as hal::Backend>::Image,
+        <back::Backend as hal::Backend>::ImageView,
+    )>,
+    extent: hal::window::Extent2D,
+) -> Vec<<back::Backend as hal::Backend>::Framebuffer> {
+    let mut swapchain_framebuffers: Vec<<back::Backend as hal::Backend>::Framebuffer> = Vec::new();
+
+    for (_, image_view) in frame_images.iter() {
+        // why is extent.depth the same as number of layers in vkSwapchainCreateInfo?
```

Comment:
These should both just be `1` when not multi/stereo surface, there is some more information in the Vulkan specification about the cases when you'd want these (i.e. VR). Is there anything specifically you were wondering about it?

Reply:
Well, I am not confused about the number of layers from the Vulkan side, rather I do not understand why the depth field of extent is used to mean the same thing as the number of layers per Vulkan. Here is how I understand it so far: extent represents "space on the screen". Screens aren't 3D, so the depth value thus represents layers used for multi/stereo surfaces, while width/height represent "typical" 2D space.

Reply:
Oh I see, I think the naming may be the confusing part here. The `Extent` struct is reused here just to provide three dimensions – the third dimension is not necessarily a "real" depth.

For example, this could be represented by a separate struct type (maybe something called `Extent2DWithLayers` with `width`, `height`, and `layers`), but the same `Extent` struct is simply reused here with `depth` to represent `layers` instead.